### PR TITLE
Prevent circular recursion due to bundle miss

### PR DESCRIPTION
**Pull request by SergeAstapov**

### CODE REVIEW

File: packages/ember-engines/addon-test-support/index.js [hunk 1]
```diff
@@ -54,6 +55,10 @@ async function loadEngine(engineName) {
     let assetLoader = owner.lookup('service:asset-loader');
 
     await assetLoader.loadBundle(engineName);
+
+    // ownerHasEngine does registry lookup and caches "miss".
+    // With register() we tell registry that module is now available.
+    owner.register(`engine:${engineName}`, require(`${engineName}/engine`).default);
```
Comment (villander) [inline packages/ember-engines/addon-test-support/index.js:61]: Why is it necessary? I didn't understand yet what was causing the issue.

File: packages/ember-engines/addon-test-support/index.js [hunk 2]
```diff
@@ -25,7 +26,7 @@ export async function setupEngine(hooks, engineName) {
 }
 
 function ownerHasEngine(owner, engineName) {
-  return Boolean(owner.factoryFor(`engine:${engineName}`));
+  return owner.hasRegistration(`engine:${engineName}`);
```
Comment (villander) [inline packages/ember-engines/addon-test-support/index.js:29]: Why this change is necessary?

Comment (SergeAstapov) [reply packages/ember-engines/addon-test-support/index.js:29]: @villander this doing exactly the same thing, except with `factoryFor` it's two step: you get a factory class and then cast to `boolean` via `Boolean` function.

with `hasRegistration` you get boolean right away

File: packages/ember-engines/addon-test-support/index.js (hunk 1 above)
Comment (SergeAstapov) [reply packages/ember-engines/addon-test-support/index.js:61]: @villander I think this test messes up things https://github.com/villander/ember-engines/blob/master/packages/ember-engines/tests/acceptance/lazy-routeable-engine-test.js#L20

I'll try to see if we change that test setup instead and or use another API rather than `owner.hasRegistration`/`owner.factoryFor` to check if engine is available

File: packages/ember-engines/addon-test-support/index.js (hunk 2 above)
Comment (villander) [reply packages/ember-engines/addon-test-support/index.js:29]: Thanks!

**Review by villander — APPROVED** [review]
Thank you so much!
